tox.ini

```diff
@@ -0,0 +1,123 @@
+[tox]
+env_list =  # Environments to run when called with no parameters.
+    lint-{black,ruff,pyright,shellcheck,codespell,docs}
+    test-{py38,py310,py311}
+minversion = 4.5
```

Comment:
Oh one more thing — if you set this to 4.6 you can further reduce CI noise with `--no-list-dependencies` https://github.com/tox-dev/tox/pull/3024

Reply:
Great advice, thanks!